packages/block-library/src/tabs/edit.js

```diff
@@ -0,0 +1,197 @@
+				} );
+			} );
+		},
+		[
+			innerTabBlocks,
+			updateBlockAttributes,
+			__unstableMarkNextChangeAsNotPersistent,
+		]
+	);
+
+	// Set the first tab as active when the editor is loaded.
+	useEffect( () => {
+		if ( innerTabBlocks?.length ) {
+			setActiveTab( innerTabBlocks[ 0 ].clientId );
+		}
+	}, [] ); // eslint-disable-line react-hooks/exhaustive-deps -- set first tab as active when the editor is loaded.
+
+	// Update active tab when selection or blocks change.
+	useEffect( () => {
+		const hasActiveTab =
+			innerTabBlocks &&
+			innerTabBlocks.some( ( block ) => block.attributes.isActive );
+
+		if ( selectedTabClientId ) {
+			// If an inner tab block is selected, or its inner blocks are selected, it becomes the active tab.
+			setActiveTab( selectedTabClientId );
+		} else if ( ! hasActiveTab && innerTabBlocks?.length ) {
+			// Otherwise, if there's no active tab, default to the first inner tab.
+			setActiveTab( innerTabBlocks[ 0 ].clientId );
+		}
+	}, [ innerTabBlocks, selectedTabClientId, setActiveTab ] );
+
+	/**
+	 * Cache data needed for save functions:
+	 * - Labels for inner tabs to generate tab list
+	 * - tabIndex to save the first tab as active, independent of the editor view
+	 */
+	useEffect( () => {
+		__unstableMarkNextChangeAsNotPersistent();
+		setAttributes( {
+			innerTabs: innerTabBlocks.map( ( block ) => ( {
+				label: block.attributes.label,
+				id: block.attributes.anchor || block.attributes.slug,
+			} ) ),
+		} );
+
+		innerTabBlocks.forEach( ( block, index ) => {
+			__unstableMarkNextChangeAsNotPersistent();
+			updateBlockAttributes( block.clientId, {
+				tabIndex: index,
+			} );
+		} );
+	}, [
+		__unstableMarkNextChangeAsNotPersistent,
+		innerTabBlocks,
+		setAttributes,
+		updateBlockAttributes,
+	] );
+
+	const blockProps = useBlockProps();
+	const innerBlockProps = useInnerBlocksProps(
+		{
+			className: 'wp-block-tabs__content',
+		},
+		{
+			__experimentalCaptureToolbars: true,
+			clientId,
+			orientation: 'horizontal',
+			template: TABS_TEMPLATE,
+		}
+	);
+
+	return (
+		<div { ...blockProps }>
+			<ul className="wp-block-tabs__list" role="tablist">
+				{ innerTabBlocks.map( ( block ) => {
+					const { anchor, isActive, label, slug } = block.attributes;
+					const tabIndexAttr = isActive ? 0 : -1;
+					const tabPanelId = anchor || slug;
+					const tabLabelId = tabPanelId + '--tab';
+
+					return (
+						<li
+							key={ block.clientId }
+							className="wp-block-tabs__list-item"
+							role="presentation"
+						>
+							<a
+								aria-controls={ tabPanelId }
+								aria-selected={ isActive }
+								className={ clsx( 'wp-block-tabs__tab-label', {
+									'is-active': isActive,
+								} ) }
+								href={ '#' + tabPanelId }
+								id={ tabLabelId }
+								onClick={ () => setActiveTab( block.clientId ) }
+								role="tab"
+								tab-index={ tabIndexAttr }
+							>
+								<RichText
+									allowedFormats={ ALLOWED_FORMATS }
```

Comment:
```suggestion
									withoutInteractiveFormatting
```

Instead of maintaining the list of allowed formats we should instead leverage the `withoutInteractiveFormatting` prop. This still allows 3rd party custom rich text formats o be used. As long as they are not interactive.

Reply:
Thank you for the pointer! Implemented in https://github.com/WordPress/gutenberg/pull/63689/commits/c6e184671d55d6ecb3008a014902c87f5499d07c